src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentSymbol/DocumentSymbolEndpoint.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.CommonLanguageServerProtocol.Framework;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.DocumentSymbol;
+
+[LanguageServerEndpoint(Methods.TextDocumentDocumentSymbolName)]
+internal class DocumentSymbolEndpoint : IRazorRequestHandler<DocumentSymbolParams, SymbolInformation[]>, ICapabilitiesProvider
+{
+    private readonly ClientNotifierServiceBase _languageServer;
+    private readonly IRazorDocumentMappingService _documentMappingService;
+
+    public DocumentSymbolEndpoint(ClientNotifierServiceBase languageServer, IRazorDocumentMappingService documentMappingService)
+    {
+        _languageServer = languageServer ?? throw new ArgumentNullException(nameof(languageServer));
+        _documentMappingService = documentMappingService ?? throw new ArgumentNullException(nameof(documentMappingService));
+    }
+
+    public bool MutatesSolutionState => false;
+
+    public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, VSInternalClientCapabilities clientCapabilities)
+    {
+        serverCapabilities.DocumentSymbolProvider = new DocumentSymbolOptions()
```

Comment:
Form now, this should check `LanguageServerFeatureOptions.SingleServerSupport`, as we don't have support in VS Code, and I feel its better to be safe than sorry, if we end up inserting a new build in VS Code without coordinating this.

Or if you're really keep you could add a new LanguageServerFeatureOption for it, that defaults to false, that way VS Code is in charge.

Reply:
added a note and am using SingleServerSupport for now